source/DateLine.mc

```diff
@@ -162,5 +198,83 @@ class DateLine extends Ui.Drawable {
 			mMonthString,
 			Graphics.TEXT_JUSTIFY_LEFT | Graphics.TEXT_JUSTIFY_VCENTER
 		);
+			
+		if (!App.getApp().getProperty("HideWeekNumber")) {
+			// Draw week letter.
+			x += dc.getTextWidthInPixels(mMonthString + " ", mFont);
+			dc.setColor(gMonoLightColour, Graphics.COLOR_TRANSPARENT);
+			dc.drawText(
+				x,
+				mY,
+				mFont,
+				mWeekOfYearLetter,
+				Graphics.TEXT_JUSTIFY_LEFT | Graphics.TEXT_JUSTIFY_VCENTER
+			);
+			// Draw week number
+			x += dc.getTextWidthInPixels(mWeekOfYearLetter, mFont);
+			dc.setColor(gMonoDarkColour, Graphics.COLOR_TRANSPARENT);
+			dc.drawText(
+				x,
+				mY,
+				mFont,
+				mWeekOfYear,
+				Graphics.TEXT_JUSTIFY_LEFT | Graphics.TEXT_JUSTIFY_VCENTER
+			);
+		}
+	}
+	
+	
+	function julianDay(year, month, day) {
+	    var a = (14 - month) / 12;
+	    var y = (year + 4800 - a);
+	    var m = (month + 12 * a - 3);
+	    return day + ((153 * m + 2) / 5) + (365 * y) + (y / 4) - (y / 100) + (y / 400) - 32045;
 	}
+
+	function isLeapYear(year) {
+	    if (year % 4 != 0) {
+	        return false;
+	    } else if (year % 100 != 0) {
+	        return true;
+	    } else if (year % 400 == 0) {
+	        return true;
+	    }
+	
+	    return false;
+	}
+	
+	function isoWeekNumber(year, month, day) {
+	    var firstDayOfYear = julianDay(year, 1, 1);
+	    var givenDayOfYear = julianDay(year, month, day);
+	
+	    var dayOfWeek = (firstDayOfYear + 3) % 7; // days past thursday
+	    var weekOfYear = (givenDayOfYear - firstDayOfYear + dayOfWeek + 4) / 7;
+	
+		// week is at end of this year or the beginning of next year
+	    if (weekOfYear == 53) {
+	
+	        if (dayOfWeek == 6) {
+	            return weekOfYear;
+	        } else if (dayOfWeek == 5 && isLeapYear(year)) {
+	            return weekOfYear;
+	        } else {
+	            return 1;
+	        }
+	    }
+	
+		// week is in previous year, try again under that year
+	    else if (weekOfYear == 0) {
+	        firstDayOfYear = julianDay(year - 1, 1, 1);
+	
+	        dayOfWeek = (firstDayOfYear + 3) % 7;
+	
+	        return (givenDayOfYear - firstDayOfYear + dayOfWeek + 4) / 7;
+	    }
+	
+		// any old week of the year
+	    else {
+	        return weekOfYear;
+	    }
+	}
```

Comment:
Got this from here: https://forums.garmin.com/developer/connect-iq/f/discussion/2778/week-of-year